src/Domain/Item/ItemService.ts

```diff
@@ -0,0 +1,75 @@
+import * as dayjs from 'dayjs'
+import { inject, injectable } from 'inversify'
+import TYPES from '../../Bootstrap/Types'
+import { GetItemsDTO } from './GetItemsDTO'
+
+import { GetItemsResult } from './GetItemsResult'
+import { Item } from './Item'
+import { ItemQuery } from './ItemQuery'
+import { ItemRepositoryInterface } from './ItemRepositoryInterface'
+import { ItemServiceInterface } from './ItemServiceInterface'
+import { SaveItemsResult } from './SaveItemsResult'
+
+@injectable()
+export class ItemService implements ItemServiceInterface {
+  private readonly DEFAULT_ITEMS_LIMIT = 100000
+  private readonly SYNC_TOKEN_VERSION = 2
+
+  constructor (
+    @inject(TYPES.ItemRepository) private itemRepository: ItemRepositoryInterface,
+  ) {
+  }
+
+  async getItems(dto: GetItemsDTO): Promise<GetItemsResult> {
+    const lastSyncTime = this.getLastSyncTime(dto)
+
+    const itemQuery: ItemQuery = {
+      userUuid: dto.userUuid,
+      lastSyncTime,
+      contentType: dto.contentType,
+      deleted: lastSyncTime ? undefined : false,
+      sortBy: 'updatedAt',
+      limit: !dto.limit || dto.limit < 1 ? this.DEFAULT_ITEMS_LIMIT : dto.limit
+    }
+
+    const items = await this.itemRepository.findAll(itemQuery)
+
+    const lastSyncTimeWithPrecautiousMicrosecond = (items[items.length - 1].updatedAt + 1) / 1000000
+    const cursorToken = Buffer.from(`${this.SYNC_TOKEN_VERSION}:${lastSyncTimeWithPrecautiousMicrosecond}`, 'utf-8').toString('base64')
```

Comment:
So cursorToken should be conditional. A cursorToken should be returned only if `items.length > dto.limit`, otherwise it should be undefined.

Reply:
Also note [this comment](https://github.com/standardnotes/syncing-server/blob/develop/lib/sync_engine/2019_05_20/sync_manager.rb#L152) for cursorToken:

          # Note that we don't add 1 microsecond to the date here like we do for the sync_token.
          # This is because some items may have the exact same updated_at value, like when
          # initially saving a large amount of imported data. Adding 1 microsecond here
          # would make us miss these items on subsequent pages.

Reply:
Ah, I've based the microsecond add on the pre-revert state of develop: https://github.com/standardnotes/syncing-server/commit/c53e4f26e8a604ea803328e428422ce99b635048#diff-93ff14120c01a9761658c9a26922385b4982682e4b7c29776a48443ef8320927R19

Will apply the "more precise sync" in a different PR in that case

Reply:
@karolsojko I do like how clean your code is. I'm wondering where you think a comment like the one above should live, if at all? Perhaps in some sort of documentation, if not in the code directly? I fear it may not be self-evident why we don't add 1 microsecond to the cursor_token, but we add it to the sync_token. Personally this is something I always forget until I read the comment.

Reply:
I'm not a fan of comments in code - we should strive to take extra effort to make the code readable so either giving long names to variables (if that helps) for example: `lastSyncTimeWithPrecautiousMicrosecond` or naming tests like I with: `it('should return a cursor token if there are more items than requested with limit')`